src/Servers/Kestrel/Core/src/Internal/Http2/Http2FrameWriter.cs

```diff
@@ -439,24 +439,29 @@ private async ValueTask<FlushResult> WriteDataAsync(int streamId, StreamOutputFl
                             dataLength = 0;
                         }
 
+                        if (_minResponseDataRate != null)
+                        {
+                            _timeoutControl.BytesWrittenToBuffer(_minResponseDataRate, _unflushedBytes);
```

Comment:
We should add a comment indicating that we call into ITimeoutControl.BytesWrittenToBuffer before flushing the output pipe for testing reasons.